pkg/hs/urbit-king/lib/Urbit/King/CLI.hs

```diff
@@ -44,6 +46,19 @@ data Opts = Opts
     }
   deriving (Show)
 
+-- | Options for the logging subsystem.
+data Log = Log
+    { lTarget  :: Maybe (LogTarget FilePath)
+    , lLevel   :: LogLevel
+    }
+  deriving (Show)
+
+data LogTarget a
```

Comment:
Why is this `a` instead of just putting `FilePath` in `LogFile`? Are you planning on reusing this somewhere else?

Reply:
Yeah, actually I do, but it's kind of gross. in logTarget in Main.hs, I take a Maybe (LogTarget FilePath) to a LogTarget (Maybe FilePath). In the argument, Nothing encodes "log to the default location"; in the result, LogFile Nothing encodes "log to the default file (which you have to run an IO action to find)"